components/test_pd/src/mocker/service.rs

```diff
@@ -353,4 +355,31 @@ impl PdMocker for Service {
         resp.set_header(header);
         Some(Ok(resp))
     }
+
+    fn update_service_gc_safe_point(
+        &self,
+        req: &UpdateServiceGcSafePointRequest,
+    ) -> Option<Result<UpdateServiceGcSafePointResponse>> {
+        let val = self
+            .service_gc_safepoint
+            .fetch_update(Ordering::SeqCst, Ordering::SeqCst, |v| {
```

Comment:
Doesn't the mock pd distinguish service safe points of different services?

Reply:
Nope, because I'm only going to test the function `check_update_service_safe_point_resp`, so I think perhaps we don't need a full-featured PD mock?

Reply:
It perhaps doesn't matter much though, I think it might be able to avoid some possible confusion or misunderstanding if someone want to do more stuff with it if we implement it completely.

Reply:
Perhaps I can add some assertions that make sure the user won't misuse this. Implementing a full-featured simulation is also an option (and I guess there won't be much more codes needed than the former), but I still prefer the assert method, because we are in fact not going to test that part and won't add code to test the full-featured implementation (who tests test code?), if the full-featured implementation was wrong, it will be even more confusing for the future users.

Reply:
(Well, I have just noticed that implementing the full-featured version is far easier than make a sound assertion...)